packages/composite-checkout-wpcom/src/components/form-field-annotation.tsx

```diff
@@ -67,7 +68,7 @@ type LabelProps = {
 	theme?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
 };
 
-const Label = styled.label< LabelProps >`
+const Label = withTheme( styled.label< LabelProps >`
```

Comment:
This _seems_ ok, but isn't this supposed to happen automatically? The [docs](https://emotion.sh/docs/theming) say that if the component is wrapped in a theme provider, you can access the theme with `props.theme`. I want to look into the tests to see if maybe the theme provider there stopped working somehow before we change code in the actual components.

Reply:
The `ThemeProvider` is being used here: https://github.com/Automattic/wp-calypso/blob/master/packages/composite-checkout-wpcom/test/form-field-annotation.js#L40

and it includes `textColor` which is what's in the error message: https://github.com/Automattic/wp-calypso/blob/master/packages/composite-checkout-wpcom/test/form-field-annotation.js#L26

Reply:
Yes, the issue isn't that the test is wrong; it's that the library isn't behaving as documented *when being tested*, for some reason. Perhaps because of an odd interaction with JSDOM? The `theme` prop makes it to the `styled` components as `{}` instead of the object you point to.

All I can say is that this is a problem with the library, not `composite-checkout`, because even a trivial test based on a modified sample from their docs fails with the same error:

```js
/**
 * @jest-environment jsdom
 */
import React from 'react';
import renderer from 'react-test-renderer';
import serializer from 'jest-emotion';
import styled from '@emotion/styled';
import { ThemeProvider } from 'emotion-theming';

const theme = {
	colors: {
		textColor: 'blue',
	}
};

expect.addSnapshotSerializer( serializer );

test( 'renders with correct styles', () => {
	const H1 = styled.h1`
		color: ${props => props.theme.colors.textColor};
	`;

	const tree = renderer
		.create(
			<ThemeProvider theme={ theme }>
				<H1 />
			</ThemeProvider>
		)
		.toJSON();

	expect( tree ).toMatchSnapshot();
} );
```

```
TypeError: Cannot read property 'textColor' of undefined
```

Reply:
Alright, looks like I'm getting closer to the issue. It has to do with where modules get loaded from; in this case, if I delete the `node_modules` directory from the `composite-checkout-wpcom` package, it looks like I can get rid of the issue. Investigating.

Reply:
Looks like the issue was due to duplicate dependencies, which let to the same module being loaded from multiple places, and thus not being able to match consumer with provider. Wow, monorepos can be such a mess...

I reverted the changes to the component, and it should all still work!